### PR TITLE
cleaning up main.go by separating code into appropriate files

### PR DESCRIPTION
**Pull request by nvcastelli**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 	"fmt"
 	"log"
 	"net/http"
-	"regexp"
+	//"regexp"
```
Comment (jraximus) [inline main.go]: remove this since it's in other file, no reason to keep commented stuff.

File: server.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	//"database/sql"
```
Comment (jraximus) [inline server.go]: same as above

File: server.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	"fmt"
+	"log"
+	"net/http"
+
+	//"github.com/gorilla/websocket"
+)
+
+func demo() {
+    fmt.Println("READING FRMO ANOTHER FILE")
```
Comment (jraximus) [inline server.go]: is this used to verify package idea works?

File: server.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		//Grab the next message from the broadcast channel
+
+		msg := <-broadcast
+
+		regExMesg := msg.Message
+
+		cmds := []string{"!!age;", "!!name;", "!!hello;"}
+
+		for _, cmds := range cmds {
+			regExMesg = replaceCommands(regExMesg, cmds)
```
Comment (jraximus) [inline server.go]: this part is good.

File: server.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+			log.Printf("error: %v", err)
+			delete(clients, ws)
+			break
+		}
+
+		broadcast <- msg
+	}
+}
+
+func handleMessages() {
```
Comment (jraximus) [inline server.go]: I think this is okay, but I was thinking this needs to be modified to be a little different, it's doing too much atm. Let me sit on this idea and I will comment further.

Think of it this way, it should be easy to unit test, if it isn't, it probably should be broken up a bit.

File: server.go (hunk 3 above)
Comment (nvcastelli) [reply server.go]: Yes, as you can tell a lot of this has debugging info in it, I'll go through and remove what I added that's superfluous.

**Pull request merged**